### DISCUSSION TITLE
Support Variant Groups

### DISCUSSION THREAD

**Discussion by oski646** [tailwindlabs/tailwindcss#5091] [Ideas]
It would be great to have something like this: `focus:(outline-none ring-2 ring-purple-600 border-transparent)` instead of `focus:outline-none focus:ring-2 focus:ring-purple-600 focus:border-transparent`.

I've seen this feature here: https://github.com/windicss/vite-plugin-windicss

**Comment by simonswiss** [comment]
I cannot put my hands on it, but I know Adam responded to this question a couple of times. This syntax wouldn't work with the JIT engine, since a _space_ is technically a delimiter between two classes, and cannot be used inside a class.

That's the same reason why you can't use any space in a JIT arbitrary value, when doing things like `calc()` or fancy grid template columns.

Perhaps a custom delimiter would allow this, but I think there was a reason why this wasn't as straightforward as it sounds.

I haven't worked on the JIT internals myself, and I am not quite sure, to be honest. 

From a "consumer" standpoint, this could be quite useful for things like `before` and `after` pseudo-elements, where you typically apply a bunch of classes.

**Reply by oski646** [reply]
I get it. Thanks for the response! 🎉
